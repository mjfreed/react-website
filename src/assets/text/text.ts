export const dummyText: string = `
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean nisi lacus, pellentesque nec nunc at, tempor tristique nulla. Duis ornare eget nunc id euismod. Nullam efficitur tellus eu mi porttitor, a bibendum nibh elementum. Donec varius ut ligula eget laoreet. Quisque justo mauris, pellentesque eget sagittis sed, posuere sed lectus. Nunc congue urna vel sapien sagittis, vel tincidunt massa interdum. Aliquam lacinia nisl id erat feugiat egestas. Maecenas porta egestas libero, sed vehicula libero placerat nec. 
`

export const workplaceBlurb: string = `
    I have been in the industry since August of 2018, the summer after my graduation. I got my start as a Full Stack Software Engineer using React and TypeScript
    in the web client and express.js with TypeScript in the web server. My first job was a 6 month contract with a start up located in Bellevue, Washington creating
    a logistics web app from the ground up. It was extremely fast-paced and I learned a lot during my tenure, not only about software development but also about
    business requirements and what it means to present a product in front of prospective customers. At my current company I spent my first year and a half still
    working as a Full Stack Software Engineer with React, express.js and TypeScript but transferred to a different team due to my curiosity about the different avenues
    of software engineering. For the past couple of years now my work has mainly focused on the back-end/infrastructure side of things, mostly revolving around AWS as
    well as some dev tooling that abstracts deployments of services to AWS to enhance developer productivity. Another component of my work has revolved around evolving
    our CI/CD pipeline with Jenkins to ensure each project is built and deployed to the specification of the teams who own and develop them. Recently, I have earned my
    AWS Certified Developer - Associate certification to elevate myself professionally, make more informed decisions when performing tasks at work and show my superiors
    and potential employers that I am dedicated to my craft and am serious about meeting personal and professional goals.
`;
export const outsideWorkplaceBlurb: string = `
    When it comes to life outside the workplace I'm a person of many hobbies and interests. My main hobby is gaming (surprise!). I'm an avid gamer of all kinds - video games,
    tabletop games, card games and physical games like golf and disc golf. I play video games on all platforms including PC, Nintendo and PlayStation. My favorite kinds of
    games are usually single player narrative games (Zelda, The Last of Us, and God of War 4 just to name a few) but I do enjoy queueing competitive multiplayer games with my friends
    from time to time (Counter-Strike is a favorite of mine). Some of my favorite board games include King of Tokyo, One Night Werewolf, Scythe and Bohnanza (or "Beans" as I like to call it),
    again, just to name a few. When I'm not gaming you can usually find me out at restaraunts trying new foods, golfing/disc golfing with my friends or watching the latest shows/movies on Netflix
    or whatever streaming platform has the thing I want to watch. Another hobby of mine is learning about cooking/how to cook. I enjoy learning about cooking and how it works from a 
    scientific perspective. Some of the people I follow are J. Kenji Lopez-Alt and Dan Souza. If you haven't seen them and enjoy cooking, check them out! Lastly, but certainly most
    dear to me, is that I'm something of a cat dad. My girlfriend introduced me to her cat (now our cat) years ago and now it's nearly impossible to separate us. I always thought people
    who were obsessed with cats was just something you saw in movies, but lo and behold here I am.
`;
