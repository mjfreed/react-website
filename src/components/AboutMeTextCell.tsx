import React from "react";
import { Container, Grid, Header } from "semantic-ui-react";

interface IAboutMeTextCellProps {
    header: string;
    bodyText: string;
}

const AboutMeTextCell: React.FC<IAboutMeTextCellProps> = ({ header, bodyText }) => {
    return (
        <Grid.Column>
            <Header style={{ fontSize: "3em" }} textAlign="center">
                {header}
            </Header>
            <Container text>{bodyText}</Container>
        </Grid.Column>
    );
};

export default AboutMeTextCell;
