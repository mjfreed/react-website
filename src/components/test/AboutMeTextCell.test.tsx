import { render } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router";
import { dummyText } from "../../assets/text/text";
import AboutMeTextCell from "../AboutMeTextCell";

describe("AboutMePhotoCell.tsx", () => {
    it("renders without crashing", () => {
        const { container } = render(
            <MemoryRouter>
                <AboutMeTextCell header="Hello, World!" bodyText={dummyText} />
            </MemoryRouter>
        );

        expect(container).toMatchSnapshot();
    });
});
