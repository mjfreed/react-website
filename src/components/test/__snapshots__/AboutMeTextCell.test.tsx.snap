// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutMePhotoCell.tsx renders without crashing 1`] = `
<div>
  <div
    class="column"
  >
    <div
      class="ui center aligned header"
      style="font-size: 3em;"
    >
      Hello, World!
    </div>
    <div
      class="ui text container"
    />
  </div>
</div>
`;
